src/manage_sql_configs.c

```diff
@@ -1985,6 +1985,111 @@ find_config_with_permission (const char* uuid, config_t* config,
   return find_resource_with_permission ("config", uuid, config, permission, 0);
 }
 
+/**
+ * @brief Gets an NVT preference by id or by name.
+ *
+ * Note: This currently only gets the fields needed by create_config.
+ *
+ * @param[in]  nvt_oid    OID of the NVT the preference belongs to.
+ * @param[in]  find_id    Preference id to find, or NULL.
+ * @param[in]  check_name Preference name to check.
+ * @param[in]  check_type Preference name to check.
+ * @param[in]  value      Value to assign to the preference.
+ *
+ * @return Newly allocated preference, freed with preference_free,
+ *          or NULL (on error or if not found).
+ */
+preference_t *
+get_nvt_preference_by_id (const char *nvt_oid,
+                          const char *find_id,
+                          const char *check_name,
+                          const char *check_type,
+                          const char *value)
+{
+  preference_t *new_pref;
+  char *full_name, *id, *name, *type, *nvt_name, *default_value, *hr_name;
+  array_t *alts;
+  gchar *quoted_oid, *quoted_id;
+  char **full_name_split;
+
+  full_name = name = type = nvt_name = default_value = hr_name = NULL;
+  alts = NULL;
```

Comment:
It's a bit confusing that alts is set here and also lower down, without being used in between.